apps/server/src/modules/course/repo/course.repo.integration.spec.ts

```diff
@@ -443,40 +443,83 @@ describe('course repo', () => {
 		});
 	});
 
-	describe('update Courses', () => {
-		describe('when user is existing', () => {
+	describe('removeUserReference', () => {
+		describe('when deleting a student from all courses', () => {
 			const setup = async () => {
 				const user = userFactory.build();
 				const otherUser = userFactory.build();
-				await em.persistAndFlush([user, otherUser]);
-				const course1 = courseEntityFactory.build({ name: 'course #1', students: [user, otherUser] });
-				const course2 = courseEntityFactory.build({ name: 'course #2', substitutionTeachers: [user, otherUser] });
-				const course3 = courseEntityFactory.build({ name: 'course #3', teachers: [user, otherUser] });
-				const course4 = courseEntityFactory.build({ name: 'course #1', students: [otherUser] });
 
-				await em.persistAndFlush([course1, course2, course3, course4]);
+				const course1 = courseEntityFactory.buildWithId({ name: 'course #1', students: [user, otherUser] });
+				const course2 = courseEntityFactory.buildWithId({ name: 'course #2', substitutionTeachers: [user, otherUser] });
+				const course3 = courseEntityFactory.buildWithId({ name: 'course #3', teachers: [user, otherUser] });
+				const course4 = courseEntityFactory.buildWithId({ name: 'course #4', students: [otherUser] });
+
+				await em.persistAndFlush([user, otherUser, course1, course2, course3, course4]);
+				em.clear();
 
 				return {
 					user,
+					otherUser,
 					course1,
 					course2,
 					course3,
+					course4,
 				};
 			};
 
-			it('should update Courses without deleted user', async () => {
+			it('should call nativeUpdate', async () => {
+				const { user } = await setup();
+
+				jest.spyOn(em, 'nativeUpdate');
+
+				await repo.removeUserReference(user.id);
```

Comment:
Please not work with spy in integration test, check if the user with this id is deleted in the in memory db.
Edit: It is already done by the next test. I prefare to remove this test.